test/unit/cpu/test-indexsets.cxx

```diff
@@ -32,9 +32,10 @@ TEST_F(IndexSetTest, InvalidSegments)
 {
   RAJA::RangeStrideSegment rs_segment(0, 4, 2);
 
-  EXPECT_NE(true, index_sets_[0].isValidSegmentType(&rs_segment));
-  EXPECT_NE(true, index_sets_[0].push_back(rs_segment));
-  EXPECT_NE(true, index_sets_[0].push_back_nocopy(&rs_segment));
+  // FIX ME push back will work here because RangeStrideSegment is a valid type in the IndexSet - need to come up with a new type to test with
```

Comment:
Yup. Since the compiler would catch any attempt to insert an invalid segment type (since seg types are known at compile time), I think the isValidSegmentType method should be removed.